PSReadLine/ReadLine.cs

```diff
@@ -174,9 +183,10 @@ internal static ConsoleKeyInfo ReadKey()
                     //   - a key is pressed
                     //   - the console is exiting
                     //   - 300ms - to process events if we're idle
-
+                    //   - processing of events is requested externally
+                    //   - ReadLine cancellation is requested externally
                     handleId = WaitHandle.WaitAny(_singleton._requestKeyWaitHandles, 300);
-                    if (handleId != WaitHandle.WaitTimeout)
+                    if (handleId != WaitHandle.WaitTimeout && handleId != 3)
```

Comment:
Can you add constants for each handleId and use those - it will make reading the code easier in the future.

Reply:
Great idea, that's a lot easier to read now.

I assumed you'd prefer I omit the constant for `_keyReadWaitHandle` as it's index isn't referenced. Let me know if that's incorrect though.